sdk/python/lib/pulumi/runtime/settings.py

```diff
@@ -92,6 +88,33 @@ def __init__(
         else:
             self.engine = None
 
+    @contextproperty("project")
+    def project(self) -> Optional[str]:
+        ...
+
+    @contextproperty("stack")
+    def stack(self) -> Optional[str]:
+        ...
+
+    @contextproperty("parallel")
+    def parallel(self) -> Optional[bool]:
+        ...
+
+    @contextproperty("dry_run")
+    def dry_run(self) -> Optional[bool]:
+        ...
+
+    @contextproperty("legacy_apply_enabled")
+    def legacy_apply_enabled(self) -> Optional[bool]:
+        ...
+
+    @contextproperty("feature_support")
+    def feature_support(self) -> Optional[dict]:
+        ...
```

Comment:
This interface feels a lot more ergonomic than the initial pass.  There's still a bit of weirdness but it feels much more on the right track